rust/connlib/tunnel/src/client.rs

```diff
@@ -1228,6 +1246,57 @@ impl ClientState {
         ControlFlow::Break(())
     }
 
+    fn handle_llmnr_dns_query(&mut self, packet: IpPacket, now: Instant) {
+        let Some(datagram) = packet.as_udp() else {
+            tracing::debug!(?packet, "Not a UDP packet");
+
+            return;
+        };
```

Comment:
Technically, we are also meant to support TCP for LLMNR queries but I didn't want to bother with that for now. Our responses never exceed the max UDP payload and we don't forward LLMNR queries to upstream resolvers because they are sent via multicast anyway and are only meant for the local network.